cassiopeia/data/utilities.py

```diff
@@ -165,4 +174,131 @@ def _compute_dissimilarity_map(cm, C, missing_state_indicator, nb_weights):
 
         return dm
 
-    return _compute_dissimilarity_map(cm, C, missing_state_indicator, nb_weights)
+    return _compute_dissimilarity_map(
+        cm, C, missing_state_indicator, nb_weights
+    )
+
+
+def sample_bootstrap_character_matrices(
+    character_matrix: pd.DataFrame,
+    prior_probabilities: Optional[Dict[int, Dict[int, float]]] = None,
+    B: int = 10,
+    random_state: Optional[np.random.RandomState] = None,
+) -> List[Tuple[pd.DataFrame, Dict[int, Dict[int, float]]]]:
+    """Generates bootstrapped character matrices from a character matrix.
+
+    Ingests a character matrix and randomly creates bootstrap samples by
+    sampling with replacement characters. Each bootstrapped character matrix,
```

Comment:
`sampling with replacement characters` => This sounds better: `sampling characters with replacement`